src/scope.js

```diff
@@ -400,19 +422,19 @@ class Scope {
     }
 
     __isClosed() {
-        return this.__left === null;
+        return this.__left === null || this.__left === undefined;
```

Comment:
I changed it back. Although the tests pass I dont want to risk introducing new bugs.